lib/Fhp/Action/SendSEPATransfer.php

```diff
@@ -60,11 +103,26 @@ protected function createRequest(BPD $bpd, ?UPD $upd)
                 . implode(', ', $supportedSchemas));
         }
 
-        $hkccs = HKCCSv1::createEmpty();
-        $hkccs->kontoverbindungInternational = Kti::fromAccount($this->account);
-        $hkccs->sepaDescriptor = $this->xmlSchema;
-        $hkccs->sepaPainMessage = new Bin($this->painMessage);
-        return $hkccs;
+        /** @var BaseSegment $hiccxs */
+        $hiccxs = $bpd->requireLatestSupportedParameters($bankparams);
+
+        $hkccx = $hiccxs->createRequestSegment();
+        $hkccx->kontoverbindungInternational = Kti::fromAccount($this->account);
+        $hkccx->sepaDescriptor = $this->xmlSchema;
+        $hkccx->sepaPainMessage = new Bin($this->painMessage);
+
+        if (!$this->singleTransfer) {
+            // Just always send the control sum (may be optional)
+            $hkccx->summenfeld = Btg::create($this->ctrlSum);
```

Comment:
This will crash if `$ctrlSum` remained null above. Either require it there already (if !$singleTransfer) or make it optional here too.

Reply:
Yes. An oversight. The message should be rejected if `$ctrlSum` cannot be set (mandatory field in the XML schema).